cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java

```diff
@@ -137,24 +146,37 @@ private boolean shouldTryAgain(Try<Publication> attemptSave, int efforts) {
     }
 
     private Try<Publication> tryPersistingInDatabase(Publication publication) {
-        return attempt(() -> resourceService.createPublicationWithPredefinedCreationDate(publication))
+        return attempt(() -> createPublicationDraft(publication))
                    .map(this::publishPublication);
     }
 
-    private void avoidCongestionInDatabase(long sleepTime) {
+    private Publication createPublicationDraft(Publication publication)
+        throws TransactionFailedException {
+        return resourceService.createPublicationWithPredefinedCreationDate(publication);
+    }
+
+    private void avoidCongestionInDatabase() {
```

Comment:
Trying to make the intention a bit more obvious. The intention of the randomized sleeping time is to avoid  a bunch of failing threads to awake all at the same time and create a new congestion